core-services/chat/src/main/java/org/silverpeas/core/chat/ChatUser.java

```diff
@@ -0,0 +1,279 @@
+package org.silverpeas.core.chat;
+
+import org.silverpeas.core.admin.user.constant.UserAccessLevel;
+import org.silverpeas.core.admin.user.constant.UserState;
+import org.silverpeas.core.admin.user.model.User;
+import org.silverpeas.core.admin.user.model.UserDetail;
+import org.silverpeas.core.chat.servers.ChatServer;
+import org.silverpeas.core.personalization.UserPreferences;
+import org.silverpeas.core.util.SettingBundle;
+import org.silverpeas.core.util.SilverpeasBundleList;
+
+import java.util.Date;
+
+/**
+ * A Silverpeas user with an account in a Chat service. The Silverpeas Chat Engine uses the services
+ * of an external chat server; so each user in Silverpeas has also an account in the chat server and
+ * hence has a unique identifier in that chat service.
+ * @author remipassmoilesel
+ */
+public class ChatUser extends UserDetail {
+
+  private final User user;
+
+  /**
+   * Gets a chat user from the specified Silverpeas unique identifier.
+   * @param userId the unique identifier of a user in Silverpeas.
+   * @return a user having an account in Silverpeas.
+   */
+  public static ChatUser getById(final String userId) {
+    return new ChatUser(User.getById(userId));
+  }
+
+  /**
+   * Gets the user behing the current request to Silverpeas and that have an account in the chat
+   * service.
+   * @return the current requester as a {@link ChatUser}.
+   */
+  public static ChatUser getCurrentRequester() {
+    return new ChatUser(User.getCurrentRequester());
+  }
+
+  /**
+   * Creates a new {@link ChatUser} from the specified Silverpeas user. It will convert the
+   * specified user to a chat user.
+   * @param user a Silverpeas user.
+   * @return a Chat service user.
+   */
+  public static ChatUser fromUser(User user) {
+    return new ChatUser(user);
+  }
+
+  private ChatUser(final User user) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this constructor to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1213)